src/search/heuristics/hm_heuristic.cc

```diff
@@ -286,6 +292,12 @@ class HMHeuristicFeature : public plugins::TypedFeature<Evaluator, HMHeuristic>
             "yes for tasks without conditional effects or axioms");
         document_property("preferred operators", "no");
     }
+
+    virtual shared_ptr<HMHeuristic> create_component(const plugins::Options &options, const utils::Context &) const override {
+        tuple parameter_tuple = tuple_cat(make_tuple(options.get<int>("m")),
+                                          Heuristic::get_heuristic_parameters_from_options(options));
+        return plugins::make_shared_from_tuple<HMHeuristic>(parameter_tuple);
```

Comment:
This is really an argument tuple, not a parameter tuple. Similarly to what I wrote above, it might make sense to encapsulate this pattern once and for all because we will likely see it 100+ times.  Here the potential value is higher because we will have mostly leaf classes.

For example, the whole thing could be:
```
return plugins::make_shared_from_flattened_arguments(
    options.get<int>("m"),
    Heuristic::get_heuristic_parameters_from_options(options)
);
```

We could even go one step further, use a uniform name for `get_heuristic_parameters_from_options`-style functions and encapsulate the base class call into a template:

```
return plugins:make_shared_from_tuple_with_inheritance<HMHeuristic, Heuristic>(
    options.get<int>("m"));
```

or even encapsulate the base class in a consistent way and make this:

```
return plugins:make_shared_from_tuple_with_inheritance<HMHeuristic>(
    options.get<int>("m"));
```

Or go even further, have a `get_parameters`-style function for the leaf classes away and have the whole function go away in cases where we just do the standard thing.